server/game/cards/12-SoW/HighHouseOfLight.js

```diff
@@ -0,0 +1,43 @@
+const StrongholdCard = require('../../strongholdcard.js');
+const { CardTypes, Players } = require('../../Constants');
+const AbilityDsl = require('../../abilitydsl.js');
+
+class HighHouseOfLight extends StrongholdCard {
+    setupCardAbilities() {
+        this.action({
+            title: 'Give a character with attachments bonus skill',
+            cost: AbilityDsl.costs.bowSelf(),
+            condition: () => this.game.isDuringConflict(),
+            target: {
+                cardType: CardTypes.Character,
+                controller: Players.Self,
+                cardCondition: card => card.isParticipating() && card.hasTrait('monk'),
+                gameAction: AbilityDsl.actions.cardLastingEffect(() => ({
+                    effect: AbilityDsl.effects.cardCannot({
+                        cannot: 'target',
+                        restricts: 'opponentsEvents'
+                    })
+                }))
+            },
+            then: context => ({
+                condition: context => this.game.currentConflict.getNumberOfCardsPlayed(context.player) >= 5,
+                gameAction: AbilityDsl.actions.selectRing({
+                    activePromptTitle: 'Choose a ring to take a fate from',
+                    ringCondition: ring => ring.fate >= 1,
+                    message: '{0} moves a fate from the {1} to {2}',
+                    messageArgs: ring => [context.player, ring, context.target],
+                    gameAction: AbilityDsl.actions.multiple([
```

Comment:
I think you want a single gameAction here: `AbilityDsl.actions.placeFate({ target: context.target })`, and you need to add `subActionProperties: ring => { origin: ring }`